smarts/env/wrappers/record_video.py

```diff
@@ -0,0 +1,163 @@
+# MIT License
+# 
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+# 
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# The MIT License
+#
+# Copyright (c) 2016 OpenAI (https://openai.com)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+import os
+from typing import Callable, Optional
+
+import gym
+from gym import logger
+from gym.wrappers.monitoring import video_recorder
+
+
+def capped_cubic_video_schedule(episode_id):
+    """Util"""
+    if episode_id < 1000:
+        return int(round(episode_id ** (1.0 / 3))) ** 3 == episode_id
+    else:
+        return episode_id % 1000 == 0
+
+
+class RecordVideo(gym.Wrapper):
```

Comment:
Currently, our `RecordVideo` class is very similar to, and thus duplicating, the latest `RecordVideo` class in gym. 

It seems like the `RecordVideo` class in gym is going to be deprecated and the functionality outsourced to external library like `MoviePy`, see https://github.com/openai/gym/blob/4d57b864f866b5f38ce8a05024184d675c897727/gym/wrappers/record_video.py#L40 .

Considering the above, it would be better for us to simply follow suit and use `MoviePy` so as to not maintain our own duplicated non-unique video recorder wrapper. This also helps to focus our repository to be leaner.

Reply:
Sure, this was just grabbing what was easily available. We can use `MoviePy`. I will defer this to a later time with an issue.

Reply:
#1478